### DISCUSSION TITLE
Docker Master Binaries stopped being updated

### DISCUSSION THREAD

**Discussion by ptolemy428** [moby/moby#43900] [General]
Hi,

I noticed the Docker master binary build has stopped updating.

https://master.dockerproject.org

The latest build was on the 27th March 2022. 

Any idea on why and how to get the daily builds running again?

I have been using the site for years to download and test the latest master branch builds. Would definitely love to get it going again.

Thanks,
Larry

**Comment by cpuguy83** [comment]
@thaJeztah

**Reply by thaJeztah** [reply]
Hm... I'd have to check why that is. I know some old GitHub / Jenkins accounts used for bots have been disabled in the docker org, so wondering if it's possibly related to that.

/cc @bsousaa perhaps this is something your could find out? (let me know if you need help on that)

**Comment by thaJeztah** [comment]
@ptolemy428 in the meantime (and not as convenient), we also build binaries for every pull request, and every pull request merged to master, so if you navigate to https://github.com/moby/moby/actions?query=branch%3Amaster, then you'll find an overview of the last builds from master;

<img width="1230" alt="Screenshot 2022-08-04 at 22 14 21" src="https://user-images.githubusercontent.com/1804568/182944259-4f00bb89-2314-4ef5-8c02-81e4a80579d5.png">

Clicking one of those builds will bring you to a page like this;

<img width="1596" alt="Screenshot 2022-08-04 at 22 13 36" src="https://user-images.githubusercontent.com/1804568/182944250-0df3d950-8cec-4c4f-92e6-195eaa00fee4.png">


And at the bottom, there's (cross)builds for various platforms (they may not all contain containerd and runc though, but the daemon and related binaries should be in there)

**Reply by crazy-max** [reply]
With https://github.com/moby/moby/pull/43529, containerd as well as other linked tools will be available for most platforms and not just `linux/amd64` if you're interested to take a look: https://github.com/moby/moby/actions/runs/2798315717

![image](https://user-images.githubusercontent.com/1951866/182946265-352743b4-7958-491b-8d91-b4e684991323.png)

**Reply by crazy-max** [reply]
Also https://github.com/rumpl/moby/pull/24 would allow us to easily provide artifacts through a Docker image.

![image](https://user-images.githubusercontent.com/1951866/182946815-6af98792-0d5a-48a7-86cb-8f5c19027efa.png)

**Reply by thaJeztah** [reply]
@crazy-max do you know if there's a "magic" permalink for "latest build on branch `X`"? (e.g. latest build from "master")?

**Reply by crazy-max** [reply]
No unfortunately workflow artifacts are not publicly available :disappointed:.

If you open https://github.com/moby/moby/actions/runs/2798315717 in a private session, links to artifacts are removed.

Hence https://github.com/rumpl/moby/pull/24 as a simple way to make them available.

**Reply by thaJeztah** [reply]
oh! 🤦 didn't realise those aren't public. It's not possible to change that? 😞

**Reply by thaJeztah** [reply]
Actually, I think they're public; at least I see the "artifacts" on https://github.com/moby/moby/actions/runs/2798315717 in a private session (not logged into github)

**Reply by thaJeztah** [reply]
OOOOH... they show, but no download link! 🙄 😞

**Reply by crazy-max** [reply]
Yep :disappointed:

**Reply by ptolemy428** [reply]
Hi guys, thank you so much for the replies. I really appreciate it. I tried to download the latest master builds following @thaJeztah's suggestion. after unzipping, it looks like most of the binaries are included except one "docker". Is this intentional? and if so, any idea where I can download the latest compatible binary for docker. Many thanks.

This is what I see after unzipping:

Archive:  ./binary.zip
   creating: binary-daemon/
  inflating: binary-daemon/containerd-shim-runc-v2  
  inflating: binary-daemon/containerd  
  inflating: binary-daemon/ctr       
  inflating: binary-daemon/docker-init  
  inflating: binary-daemon/docker-proxy  
  inflating: binary-daemon/dockerd-rootless-setuptool.sh  
  inflating: binary-daemon/dockerd-rootless.sh  
  inflating: binary-daemon/dockerd   
  inflating: binary-daemon/rootlesskit-docker-proxy  
  inflating: binary-daemon/rootlesskit  
  inflating: binary-daemon/runc      
  inflating: binary-daemon/vpnkit

**Reply by ptolemy428** [reply]
Hi all, any suggestion on where I can download the latest docker binary? many thanks.

Cheers,
LL
